spynnaker/pyNN/models/neural_projections/projection_machine_edge.py

```diff
@@ -56,8 +56,38 @@ def filter_edge(self, graph_mapper):
                 pre_hi = graph_mapper.get_slice(self.pre_vertex).hi_atom
                 post_lo = graph_mapper.get_slice(self.post_vertex).lo_atom
                 post_hi = graph_mapper.get_slice(self.post_vertex).hi_atom
-                if pre_hi < post_lo or pre_lo > post_hi:
-                    n_filtered += 1
+                # Filter edge if both are views and outside limits
+                if ((synapse_info.prepop_is_view) and
+                        (synapse_info.postpop_is_view)):
+                    prepop_lo = synapse_info.pre_population._indexes[0]
+                    prepop_hi = synapse_info.pre_population._indexes[-1]
+                    postpop_lo = synapse_info.post_population._indexes[0]
+                    postpop_hi = synapse_info.post_population._indexes[-1]
+                    if ((prepop_hi < pre_lo or prepop_lo > pre_hi) or
+                            (postpop_hi < post_lo or postpop_lo > post_hi)):
+                        n_filtered += 1
+                # Filter edge if pre-pop is outside limit and post_lo is bigger
+                # than n_pre_neurons
+                elif synapse_info.prepop_is_view:
+                    prepop_lo = synapse_info.pre_population._indexes[0]
+                    prepop_hi = synapse_info.pre_population._indexes[-1]
+                    prepop_size = synapse_info.n_pre_neurons
+                    if ((prepop_hi < pre_lo or prepop_lo > pre_hi) or
+                            (prepop_size < post_lo)):
+                        n_filtered += 1
+                # Filter edge if post-pop is outside limit and pre_lo is bigger
+                # than n_post_neurons
+                elif synapse_info.postpop_is_view:
+                    postpop_lo = synapse_info.post_population._indexes[0]
+                    postpop_hi = synapse_info.post_population._indexes[-1]
+                    postpop_size = synapse_info.n_post_neurons
+                    if ((postpop_hi < post_lo or postpop_lo > post_hi) or
+                            (postpop_size < pre_lo)):
```

Comment:
As above pre vs. post.

Reply:
Same answer as above, but swap pre- for post-

Reply:
Same again with a and b swapped above.